### PR TITLE
fix: crashing if handler is missing

### PR DESCRIPTION
**Pull request by sidhujus**
#### Which issue(s) does this change fix?

#6253 

#### Why is this change necessary?
There is currently no validation on if the handler is defined within a template

#### How does it address the issue?
Checks if the handler is defined before trying to access and assign it

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by jysheng123 — COMMENTED** [review]
I would add a unit test raising the specific error thrown without the handler present other than that LGTM

File: samcli/lib/providers/sam_function_provider.py [hunk 1]
```diff
@@ -445,6 +446,9 @@ def _build_function_configuration(
             codeuri = SamLocalStackProvider.normalize_resource_path(stack.location, codeuri)
 
         package_type = resource_properties.get("PackageType", ZIP)
+        if package_type == ZIP and resource_properties.get("Handler") is None:
```
Comment (lucashuy) [inline samcli/lib/providers/sam_function_provider.py:449]: Should we also validate for an empty handler (ie. `Handler: ""`)? If that's also an invalid case then we can include that in this check.

Comment (sidhujus) [reply samcli/lib/providers/sam_function_provider.py]: Yea thats a good call, ill add that in there as well

File: tests/unit/commands/local/lib/test_sam_function_provider.py [hunk 2]
```diff
@@ -1780,6 +1782,23 @@ def test_must_skip_non_existent_properties(self):
         self.assertEqual(expected, result)
 
 
+class TestSamFunctionProvider_build_function_configuration(TestCase):
+    def test_must_convert(self):
```
Comment (jysheng123) [inline tests/unit/commands/local/lib/test_sam_function_provider.py:1786]: Think this test name should be descriptive like test_handler_missing_exception

Comment (jysheng123) [reply tests/unit/commands/local/lib/test_sam_function_provider.py:1786]: Could also consider making another test for the emtpy handler string for maximum coverage

**Review by jysheng123 — APPROVED** [review]
Once the above comments are addressed it LGTM!

File: samcli/lib/providers/sam_function_provider.py [hunk 3]
```diff
@@ -446,7 +446,9 @@ def _build_function_configuration(
             codeuri = SamLocalStackProvider.normalize_resource_path(stack.location, codeuri)
 
         package_type = resource_properties.get("PackageType", ZIP)
-        if package_type == ZIP and resource_properties.get("Handler") is None:
+        if package_type == ZIP and (
+            resource_properties.get("Handler") is None or resource_properties.get("Handler") == ""
+        ):
```
Comment (lucashuy) [inline samcli/lib/providers/sam_function_provider.py:451]: nit: this could be changed to a false-y check (not sure if we also wanted to validate type)

File: tests/unit/commands/local/lib/test_sam_function_provider.py [hunk 4]
```diff
@@ -1781,6 +1782,23 @@ def test_must_skip_non_existent_properties(self):
         self.assertEqual(expected, result)
 
 
+class TestSamFunctionProvider_build_function_configuration(TestCase):
```
Comment (lucashuy) [inline tests/unit/commands/local/lib/test_sam_function_provider.py:1785]: Could we add/parameterize this test to also test for an empty string Handler?

**Review by lucashuy — COMMENTED** [review]
Thanks for addressing the blank string Handler, left a comment about testing

File: tests/unit/commands/local/lib/test_sam_function_provider.py (hunk 2 above)
Comment (sidhujus) [reply tests/unit/commands/local/lib/test_sam_function_provider.py]: ah yea i forgot to change the name from the test that i copied

**Review by lucashuy — APPROVED** [review]
Thanks for addressing the changes
